include/hal/library/cryptlib.h

```diff
@@ -4284,3 +4284,29 @@ bool libspdm_hkdf_sm3_256_expand(const uint8_t *prk, size_t prk_size,
                                  uint8_t *out, size_t out_size);
 
 #endif /* __BASE_CRYPT_LIB_H__*/
+
+/**
+ * Gen CSR
+ *
+ * @param[in]      hash_nid              hash algo for sign
+ * @param[in]      asym_nid              asym algo for sign
+ *
+ * @param[out]     csr_len               CSR len for DER format
+ * @param[in]      csr_pointer           For input, csr_pointer is address to store CSR.
+ * @param[out]     csr_pointer           For input, csr_pointer is address for stored CSR.
+ * @param[in]      csr_buffer_size       The size of store CSR buffer.
+ *
+ * @param[in]      requester_info        requester info to gen CSR
+ * @param[in]      requester_info_length The len of requester info
+ *
+ * @param[in]      context               Pointer to asymmetric context
+ * @param[in]      subject_name          Subject name: should be break with ',' in the middle
+ *                                       example: "C=AA,CN=BB"
+ *
```

Comment:
Please list all supported name. You cannot use "for example" here. Please try to use below in function header.

The subject name should be a string, "<name>=<value>,<name>=<value>,...,<name>=<value>".
The valid name is in {"CN", "commonName", "C", "countryName", "O", "organizationName","L", "locality","R", "OU", "organizationalUnitName", "ST", "stateOrProvinceName", "emailAddress", "serialNumber", "postalAddress", "postalCode", "dnQualifier", "title","surName", "SN","givenName","GN", "initials", "pseudonym", "generationQualifier", "domainComponent", "DC"}.

Reply:
I have tested of them and deleted three failed types.